### PR TITLE
Enable Running Multiple Trials As Part Of An Experiment

### PR DESCRIPTION
**Pull request by bonedaddy**
# Overview

Allows defining multiple trials within a single experiment definition to be ran concurrently, and extends cue definition to enable specifying libp2p peer parameters. 
# Changes

* Extends the cue experiment definition to include the ability to define the libp2p peer parameters
  * If not defined defaults are used
* Exports cue definition as a package level variable
* Disables session node updates as this is a source of major problems when running concurrent trials
  * To be honest, I have absolutely no clue what the problem is, I did gather some notes on this [here](https://github.com/RTradeLtd/p2plab/pull/8#issuecomment-635634935)
* Adds ability to connect clusters together
* Enables returning the results of an experiment as JSON to stdout, *or* in CSV format written to a file.
* Adds a helper library to the routers package to share common functionality between different routers
* Might be missing one or two things that i cant remember or easily classify as individual changes

# Notes

* While cleaning up the branch and squashing commits, @hinshun commits from https://github.com/RTradeLtd/p2plab/pull/9 were removed as being done by you. Let me know if  I should fix this to include your commit history as part of the PR
* In issue #39  one of the requirements for this was ` I also prefer if you kept the router light, and write the logic that actually runs experiment in the experiments package.`
  * I'm a bit unsure what code in the experiments router can be moved there, as it requires some subsystems only enabled in the router.
* There's still some todo comments left in the code from me, let me know if I should change these

# Possible Future Action Items

* Extend cue definition to allow specifying the ability to induce arbitrary/random failures, network delays, etc... [toxiproxy seem interesting for this](https://github.com/Shopify/toxiproxy/tree/master/client)

### CODE REVIEW

**Comment by bonedaddy** [comment]
Didnt notice the build failure, will investigate

File: labd/routers/helpers/helpers.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/Netflix/p2plab/labd/controlapi"
+	"github.com/Netflix/p2plab/metadata"
+	"github.com/Netflix/p2plab/nodes"
+	"github.com/Netflix/p2plab/pkg/httputil"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	bolt "go.etcd.io/bbolt"
+)
+
+// TODO(bonedaddy): not sure if this is the best way to go about sharing code between routers
```
Comment (hinshun) [inline labd/routers/helpers/helpers.go:16]: Leftover comment.

File: nodes/session.go [hunk 2]
```diff
@@ -33,18 +31,21 @@ func Session(ctx context.Context, ns []p2plab.Node, fn func(context.Context) err
...
+
 	eg, gctx := errgroup.WithContext(sctx)
 
 	zerolog.Ctx(ctx).Info().Msg("Starting a session for benchmarking")
 	go logutil.Elapsed(gctx, 20*time.Second, "Starting a session for benchmarking")
-
-	cancels := make([]context.CancelFunc, len(ns))
-	for i, n := range ns {
-		i, n := i, n
+	/*var cancels = make([]context.CancelFunc, 0, len(ns))
```
Comment (hinshun) [inline nodes/session.go:43]: Can you add a comment here so the context isn't lost?

**Review by hinshun — APPROVED** [review]
Small nits, otherwise lgtm.

File: nodes/session.go (hunk 2 above)
Comment (bonedaddy) [reply nodes/session.go:43]: That is definitely a good idea

**Comment by hinshun** [comment]
Thanks! :tada:

**Pull request merged**